cli/internal/fs/turbo_json.go

```diff
@@ -301,6 +300,46 @@ func (c *TaskDefinition) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// MarshalJSON deserializes JSON into a TaskDefinition
+func (c *TaskDefinition) MarshalJSON() ([]byte, error) {
```

Comment:
Quick glancing through the code I think we aren't altering `TaskDefinition`s after construction, but since non-determinism has gotten us before: we might want to sort these slices before marshalling them. If you've done some closer reading of the code to verify that the slices remain sorted after creation, then feel free to ignore this comment.

Reply:
Good call. That seems like a good idea, even just for determinism in tests, but I don't _think_ it should affect task hashes, right? 🤔

Reply:
Wrote a test first and it seemed to pass deterministically, but added a sort in anyway

Reply:
I wonder how visible this change will be. I _think_ it effectively changes the execution order of peer tasks?

None of that should be relied upon (and we should probably intentionally insert randomness into that at execution time to prevent people relying on it) but that doesn't mean people aren't implicitly relying on it right now.

Reply:
Based on my experience with test runs, the order is indeterminate already (see https://github.com/Nicoretti/prysk/issues/154, for example), so I don't think the sort is guaranteeing anything.

Also, for future readers, to be clear, `Marshal` doesn't play a part in taskGraph exectution at all. I'm guessing Nathan is commenting on sorting lists in general (and in this case the Unmarshal part)